src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -53,20 +53,48 @@ private static bool IsEnabled()
                    s_diagnosticListener.IsEnabled();
         }
 
-        private static Activity? CreateActivity(HttpRequestMessage requestMessage)
+        private static Activity? StartActivity(HttpRequestMessage request)
         {
             Activity? activity = null;
             if (s_activitySource.HasListeners())
             {
-                activity = s_activitySource.CreateActivity(DiagnosticsHandlerLoggingStrings.ActivityName, ActivityKind.Client);
+                Uri? requestUri = request.RequestUri;
+                if (requestUri?.IsAbsoluteUri == false)
+                {
+                    requestUri = null;
+                }
+
+                // The following tags might be important for making sampling decisions and should be provided at span creation time.
+                // https://github.com/open-telemetry/semantic-conventions/blob/5077fd5ccf64e3ad0821866cc80d77bb24098ba2/docs/http/http-spans.md?plain=1#L213-L219
+                KeyValuePair<string, object?>[] tags = requestUri is not null
+                    ? new KeyValuePair<string, object?>[4]
+                    : new KeyValuePair<string, object?>[1];
+
+                tags[0] = DiagnosticsHelper.GetMethodTag(request.Method);
+                if (requestUri is not null)
+                {
+                    tags[1] = new("server.address", requestUri.Host);
+                    tags[2] = new("server.port", requestUri.Port);
```

Comment:
would it return 80/443 if they were not specified?

Reply:
No, `server.port` seems to be `Required` to me.

Reply:
yep, it's required.

It seems [`Uri.Port`](https://learn.microsoft.com/en-us/dotnet/api/system.uri.port?view=net-8.0#remarks) does exactly what we need - returns default port for Uri if it's not specified. So all seems to be good.